cpp/src/jni/dataset/proto/record_batch.proto

```diff
@@ -1,55 +0,0 @@
-// Licensed to the Apache Software Foundation (ASF) under one
-// or more contributor license agreements.  See the NOTICE file
-// distributed with this work for additional information
-// regarding copyright ownership.  The ASF licenses this file
-// to you under the Apache License, Version 2.0 (the
-// "License"); you may not use this file except in compliance
-// with the License.  You may obtain a copy of the License at
-//
-//   http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing,
-// software distributed under the License is distributed on an
-// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
-// KIND, either express or implied.  See the License for the
-// specific language governing permissions and limitations
-// under the License.
-
-syntax = "proto2";
-package arrow.dataset.jni;
-
-option java_package = "org.apache.arrow.dataset.jni";
-option java_outer_classname = "RecordBatchProtos";
```

Comment:
@emkornfield
I managed to migrate to existing flatbuffers API from protobuf.

previous comment https://github.com/apache/arrow/pull/10108#discussion_r619887946